relayer/relayer.go

```diff
@@ -343,12 +348,32 @@ func (r *Relayer) RelayMessage(warpLogInfo *vmtypes.WarpLogInfo, metrics *Messag
 	r.currentRequestID++
 
 	// Update the database with the latest processed block height
-	err = r.db.Put(r.sourceBlockchainID, []byte(database.LatestProcessedBlockKey), []byte(strconv.FormatUint(warpLogInfo.BlockNumber, 10)))
-	if err != nil {
-		r.logger.Error(
-			fmt.Sprintf("failed to put %s into database", database.LatestProcessedBlockKey),
-			zap.Error(err),
-		)
+	if storeProcessedHeight {
```

Comment:
To reduce multiple indentation levels, I recommend inverting this if statement such that we return early if not storing the processed height, and otherwise continue on. 

May also want to move the logic of storing the processed height out to helper for readability.

Reply:
Made both of these changes 👍